### PR TITLE
Test adding and getting a large image file

### PR DESCRIPTION
**Pull request by asutula**
Wanted to see that the streaming of file data between client and server actually works when there are many chunks. Also learned about file chunking in IPFS and wanted to see that in action.

### CODE REVIEW

File: server/server_test.go [hunk 1]
```diff
@@ -85,6 +86,89 @@ func TestGetFile(t *testing.T) {
...
+		if err != nil {
+			t.Fatalf("failed to write chunk to buffer: %v", err)
+		}
+	}
+
+	out.Sync()
+
+	got := int32(len(buffer.Bytes()))
+	expected := refLargeFile.GetStat().GetCumulativeSize() - refLargeFile.GetStat().GetLinksSize()
+	if got != expected {
```
Comment (asutula) [inline server/server_test.go]: I'm trying to compare the size of the data that was added to the size of the data that comes back from GetFile. I can't figure out exactly what properties of the file that was added (`refLargeFile`) should be used to make that comparison, so this fails. The numbers are slightly different.

Comment (asutula) [reply server/server_test.go]: I'm writing the file I'm getting to disk as well and it's identical to the original file. So things seem to be working.

**Review by andrewxhill — APPROVED** [review]

File: server/server_test.go [hunk 2]
```diff
@@ -21,7 +22,7 @@ import (
 var (
 	client                                                     pb.IpfsLiteClient
 	stringToAdd                                                string = "hola"
-	refFile                                                    *pb.Node
+	refFile, refLargeFile                                      *pb.Node
```
Comment (andrewxhill) [inline server/server_test.go:25]:
```suggestion
	refFile, refLargeFile                                      *pb.Node
	refSize                                                           int32
```

File: server/server_test.go [hunk 3]
```diff
@@ -85,6 +86,89 @@ func TestGetFile(t *testing.T) {
...
+	if err != nil {
+		t.Fatalf("failed to AddLargeFile: %v", err)
+	}
+
+	file, err := os.Open("test-data/IMG_2293.jpeg")
+	if err != nil {
+		t.Fatalf("failed to open file: %v", err)
+	}
+	defer file.Close()
+
```
Comment (andrewxhill) [inline server/server_test.go]:
```suggestion
	fi, err := file.Stat()
	if err != nil {
		t.Fatalf("failed to stat file: %v", err)
	}
	refSize = int32(fi.Size())
```

File: server/server_test.go [hunk 4]
```diff
@@ -85,6 +86,89 @@ func TestGetFile(t *testing.T) {
...
+		_, err = buffer.Write(resp.GetChunk())
+		if err != nil {
+			t.Fatalf("failed to write chunk to buffer: %v", err)
+		}
+	}
+
+	out.Sync()
+
+	got := int32(len(buffer.Bytes()))
+	expected := refLargeFile.GetStat().GetCumulativeSize() - refLargeFile.GetStat().GetLinksSize()
```
Comment (andrewxhill) [inline server/server_test.go]:
```suggestion
```

remove this line

File: server/server_test.go (hunk 1 above)
Comment (andrewxhill) [inline server/server_test.go]:
```suggestion
	if got != refSize {
```

File: server/server_test.go [hunk 5]
```diff
@@ -85,6 +86,89 @@ func TestGetFile(t *testing.T) {
...
+	if got != expected {
+		t.Fatalf("wanted %d but got: %d", expected, got)
```
Comment (andrewxhill) [inline server/server_test.go]:
```suggestion
		t.Fatalf("wanted %d but got: %d", refSize, got)
```

**Comment by asutula** [comment]
Thanks for the changes @andrewxhill ;)

**Pull request merged**